web/components/FcDrawerViewData.vue

```diff
@@ -419,11 +401,11 @@ export default {
       this.studySummary = studySummary;
       this.studyTotal = studyTotal;
 
-      if (this.location === null
-          || location.centrelineId !== this.location.centrelineId
-          || location.centrelineType !== this.location.centrelineType
-          || location.description !== this.location.description) {
-        this.setLocation(location);
+      const { s1 } = this;
+      if (s1 !== s1Next) {
+        const locationMap = await getLocationsByFeature(features);
```

Comment:
this suggests that `getLocationsByFeature` should *not* return a `Map`, but instead an `Array` - we can always build a map out of an `Array`, e.g. as

```
const locationsMap = new Map(
  locations.map(location => [centrelineKey(location), location]),
);